Task 3

Sophiliya opened the pull request.

Sophiliya committed:
Add files via upload

Task #3

Sophiliya committed:
Delete task_3_1.rb

Sophiliya committed:
Delete task_3_2.rb

Sophiliya committed:
Add files via upload

Task #3

Sophiliya committed:
Update task_3_1.rb

Исправила

Sophiliya committed:
Update task_3_2.rb

Исправила

Sophiliya committed:
Update task_3_3.rb

Исправила

dimasumskoy commented on task_3_2.rb:
```diff
@@ -0,0 +1,32 @@
+array = []
```
Называть переменные лучше осмысленно. Тут у вас будут храниться стороны треугольника, поэтому логичнее назвать переменную sides, например.

dimasumskoy commented on task_3_2.rb:
```diff
@@ -0,0 +1,32 @@
+array = array.sort!
+
+f_rectangular = array[2]**2 == array[0]**2 + array[1]**2
+f_isosceles = ( array[0] == array[1] && array[1] != array[2] ) || ( array[0] != array[1] && array[1] == array[2] )
```
Сложно с первого взгляда разобраться, что тут происходит. Выражение излишне сложное.
Всегда стремитесь делать все как можно проще, даже если от этого увеличится кол-во строк кода. Тем более по-началу это не так критично.

Тут я бы вам посоветовал ввести переменные. Например: 
`hypotenuse = array[2]`
`second_side = array[1]`
`third_side = array[0]`

Далее вынести выражение вычисления гипотенузы в отдельную переменную: `hypotenuse ** 2 == second_side ** 2 + third_side ** 2`, и потом в условии определять является ли треугольник равносторонним, прямоугольным, прямоугольным и равнобедренным, или не прямоугольным. Используя переменные, код будет выглядеть намного чище и понятнее.

dimasumskoy commented on task_3_2.rb:
```diff
@@ -0,0 +1,32 @@
+  
+elsif f_rectangular && !f_isosceles
+    puts 'Rectangular!'
+  
```
В условиях такие пустые строки излишни. Они не прибавляют читаемости, а просто только удлиняют код.

dimasumskoy commented on task_3_3.rb:
```diff
@@ -0,0 +1,23 @@
+  puts "x1 = #{x1} and x2 = #{x2}."
+  
+elsif d == 0
+  x = -b/(2*a)
```
Тут забыли пробелы (:

dimasumskoy commented on task_3_3.rb:
```diff
@@ -0,0 +1,23 @@
+b = gets.chomp.to_f
+c = gets.chomp.to_f
+
+d = b ** 2 - 4 * a * c
```
Для лучшей читаемости лучше написать так: `d = b ** 2 - (4 * a * c)`

Sophiliya committed:
Update task_3_2.rb

Изменила название массива; добавила переменные для сторон треугольника; убрала пустые строки в условиях.

Sophiliya committed:
Update task_3_3.rb

Добавила скобки (4 * b * c) и пробелы -b / ( 2 * a )
